api/src/main/java/org/apache/iceberg/catalog/Namespace.java

```diff
@@ -19,19 +19,19 @@
 
 package org.apache.iceberg.catalog;
 
-import java.util.Arrays;
 import org.apache.iceberg.relocated.com.google.common.base.Joiner;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.immutables.value.Value;
 
 /**
  * A namespace in a {@link Catalog}.
  */
-public class Namespace {
-  private static final Namespace EMPTY_NAMESPACE = new Namespace(new String[] {});
+@Value.Immutable
+public abstract class Namespace {
```

Comment:
JSON serde can be done as shown below once #3424 would need this:
```
@JsonSerialize(as = ImmutableNamespace.class)
@JsonDeserialize(as = ImmutableNamespace.class)
```

The same applies for `TableIdentifier`